### PR TITLE
planner/cascades: add rule transformation rule InjectProjectionBelowAgg

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Problem Summary:
Inject proj below agg in AggImpl.AttachChildren will occur error because it will modify AggFunc and all Impl share the same  logical plan.

### What is changed and how it works?

This PR add a new rule InjectProjectionBelowAgg which has the same logic with InjectProjBelowAgg().

### Check List 

Tests 

- Unit test
- Integration test

### Release note
- add a new rule InjectProjectionBelowAgg which has the same logic with InjectProjBelowAgg().

### CODE REVIEW

**Comment by SeaRise** [comment]
emm....I'm confused about the conflict.

**Comment by LittleFall** [comment]
> emm....I'm confused about the conflict.

Excuse me, what confused you?

**Comment by SeaRise** [comment]
> > emm....I'm confused about the conflict.
> 
> Excuse me, what confused you?

This branch conflicts with master.
Never mind。

**Comment by LittleFall** [comment]
This conflict may be caused by force push. I have just merge it.
There are some tests failed after merge master, PTAL.
@SeaRise

File: planner/cascades/transformation_rules.go [hunk 1]
```diff
@@ -2222,6 +2225,122 @@ func (r *InjectProjectionBelowTopN) OnTransform(old *memo.ExprIter) (newExprs []
...
+
+	projSchemaCols := make([]*expression.Column, 0, len(copyFuncs)+len(agg.GroupByItems))
+	projExprs := make([]expression.Expression, 0, cap(projSchemaCols))
+
+	for _, f := range copyFuncs {
+		for i, arg := range f.Args {
+			switch expr := arg.(type) {
+			case *expression.Constant:
+				continue
+			case *expression.Column:
```
Comment (francis0407) [inline planner/cascades/transformation_rules.go:2288]: Good enhancement!

**Comment by SeaRise** [comment]
> This conflict may be caused by force push. I have just merge it.
> There are some tests failed after merge master, PTAL.
> @SeaRise

think you, I has fix the tests

**Comment by francis0407** [comment]
@SeaRise Please resolve the conflicts. I think it will pass the common-test after merging the master into this branch.

@lzmhhh123 Could you please help merge this PR today?

**Comment by lzmhhh123** [comment]
> @SeaRise Please resolve the conflicts. I think it will pass the common-test after merging the master into this branch.
> 
> @lzmhhh123 Could you please help merge this PR today?

OK.

**Comment by francis0407** [comment]
PTAL @lzmhhh123

**Comment by francis0407** [comment]
@lzmhhh123 PTAL

**Pull request merged**